scripts/opendataafrica/download_folder/download.sh

```
@@ -0,0 +1,117 @@
+  for dataset in "${datasets[@]}"; do
+    download_dataset "$dataset"
+    if [ $? -eq 0 ]; then
+      local json_output_file="$OUTPUT_XML_JSON_DIR/${dataset}.json"
+      local csv_output_file="$OUTPUT_CSV_DIR/${dataset}.csv"
+      if python3 xml_to_json.py --input_xml="$OUTPUT_XML_JSON_DIR/${dataset}.xml" --output_json="$json_output_file"; then
+        echo "Successfully converted ${dataset}.xml to ${dataset}.json"
+        # Corrected call to json_to_csv.py with flags and output directory
+        if python3 json_to_csv.py --input_json="$OUTPUT_XML_JSON_DIR" --output_csv="$OUTPUT_CSV_DIR"; then
+          echo "Successfully converted JSON files in $OUTPUT_XML_JSON_DIR to CSV files in $OUTPUT_CSV_DIR"
+        else
+          echo "Error: Failed to convert JSON files to CSV files."
+        fi
+      else
+        echo "Error: Failed to convert ${dataset}.xml to ${dataset}.json"
+      fi
+    else
+      echo "Error: Failed to download dataset $dataset"
+    fi
+  done
+}
+
+function download_all_datasets_for_country() {
+  if [ -z "$1" ]; then
+    echo "No country specified. Exiting"
+    exit 1
+  fi
+  COUNTRY="$1"
+
+  if [ -z "$2" ]; then
+    echo "No output CSV folder specified. Exiting"
+    exit 1
+  fi
+  OUTPUT_CSV_DIR="$2"
+
+  echo "Making output directory for XML/JSON: $OUTPUT_XML_JSON_DIR"
+  mkdir -p "$OUTPUT_XML_JSON_DIR"
+
+  echo "Making output directory for CSV: $OUTPUT_CSV_DIR"
+  mkdir -p "$OUTPUT_CSV_DIR"
+
+  download_key_families
+  if [ -f "$OUTPUT_XML_JSON_DIR/key_family.xml" ]; then
+    xml_to_json "$OUTPUT_XML_JSON_DIR/key_family.xml" "$OUTPUT_XML_JSON_DIR/key_family.json"
+    if [ -f "$OUTPUT_XML_JSON_DIR/key_family.json" ]; then
+      extract_key_families
+      echo "Datasets = ${datasets[@]}"
+      echo "Dataset length = ${#datasets[@]}"
+      download_all_datasets
+    else
+      echo "Error: key_family.json not found. Conversion failed."
+    fi
+  else
+    echo "Error: key_family.xml not found. Download failed."
```

For all errors that should abort processing pls 'exit 1' so autorefreah too about rather than assume a series data doesn't exist for country

---

Resolved